config/config_test.go

```diff
@@ -26,6 +29,22 @@ import (
 	"github.com/influxdata/telegraf/plugins/parsers/json"
 )
 
+func TestReadBinaryFile(t *testing.T) {
+	os.Chdir("../")
+	tmpdir := t.TempDir()
+	binaryFile := filepath.Join(tmpdir, "custom_builder")
+	cmd := exec.Command("go", "build", "-o", binaryFile, "./tools/custom_builder")
+	var outb, errb bytes.Buffer
+	cmd.Stdout = &outb
+	cmd.Stderr = &errb
+	err := cmd.Run()
+	require.NoError(t, err, fmt.Sprintf("stdout: %s, stderr: %s", outb.String(), errb.String()))
+	c := NewConfig()
+	err = c.LoadConfig(binaryFile)
+	require.ErrorContains(t, err, "provided config is not a text file")
+	require.Error(t, err)
```

Comment:
Shouldn't these two be swapped, I think otherwise, the last one doesn't make much sense?